language/tools/move-package/tests/test_sources/diamond_problem_dep_incorrect_override_occupied/Move.resolved

```diff
@@ -0,0 +1,3 @@
+Failed to resolve dependencies for package 'Root': Resolving dependencies for package 'B': Conflicting dependencies found:
```

Comment:
I agree that this would be ideal but it's currently impossible... It's due to the restriction that when we process an overridden dependency we do not add this dependency to the set of overrides:
```
        // It's also pretty important that we do not extend overrides with the override being
        // currently processed as we have no way of knowing if it's the correct one (dominating all
        // package "uses"). If we did, we could override an existing (non-overridden) entry in the
        // graph with an incorrect override without being able to detect it.
```
In other words, in this case, there isn't even a notion of override when processing this particular pair of conflicting packages...

Reply:
I have updated the doc comment to explain the restriction leading to different error messages a bit better:
```
        // It's also pretty important that we do not extend overrides with the override being
        // currently processed. The reason for it is that in order to detect incorrect overrides
        // (such that do not dominate all package "uses") we rely on the package being reachable via
        // different paths:
        // - if it's reached via an overridden path again for the same override, it's OK
        // - if it's reached via an overridden path again for a different override, it's an error
        // - if it's reached via a non-overridden path, it's an error (insufficient override)
        //
        // While the first type of error could still be detected if we injected the currently
        // processed override into the overrides set, the second one would not.  Consider
        // diamond_problem_dep_incorrect_override_occupied example (in tests) to see a situation
        // when a non-overridden path is chosen first to insert the package and then insufficient
        // override could be considered correct if we injected it into the overrides set (as we will
        // not have another path to explore that would disqualify it).

```